src/arch/riscv64/mod.rs

```diff
@@ -8,4 +10,10 @@ impl Riscv64 {
     }
 }
 
-impl Architecture for Riscv64 {}
+impl Architecture for Riscv64 {
+    #[naked]
+    #[no_mangle]
+    unsafe extern "C" fn _start() -> ! {
```

Comment:
Isn't this going to be an issue once we add multiple architectures? i.e multiple definitions for function `_start`? If not then all good, otherwise, we can have a start function akin to the following:

```rust
#[naked]
#[no_mangle]
unsafe extern "C" fn _start() -> ! {
    cfg_if! {
        if #[cfg(target_arch = "riscv64")] {
            // which does the inline asm, and we can add a 'startup' function
            // to the Architecture trait for example
            return RiscV::startup();
        } else {
            // oooops
        }
    }
}
```
We can even call `k_main` at the end of this function if this is a possibility for all architectures, not sure about that.

Reply:
Using conditional compilation, there will only be on struct implementing the `Architecture` trait. There will be only one _start function then